diff --git a/node_modules/storybook/dist/channels/index.js b/node_modules/storybook/dist/channels/index.js
index 64b1437..dcd53e2 100644
--- a/node_modules/storybook/dist/channels/index.js
+++ b/node_modules/storybook/dist/channels/index.js
@@ -117,7 +117,7 @@ var { document: document2, location } = globalThis, KEY = "storybook-channel", C
     );
     return frames.length ? (this.buffer.length && this.flush(), frames.forEach((f) => {
       try {
-        f.postMessage(data, "*");
+        f.postMessage(data, globalThis.location ? globalThis.location.origin : "*");
       } catch {
         logger2.error("sending over postmessage fail");
       }
@@ -157,6 +157,12 @@ var { document: document2, location } = globalThis, KEY = "storybook-channel", C
     ).flatMap((e) => e.contentWindow ? [e.contentWindow] : []) : globalThis.parent ? [globalThis.parent] : [];
   }
   handleEvent(rawEvent) {
+    if (
+      rawEvent.origin &&
+      globalThis.location &&
+      rawEvent.origin !== globalThis.location.origin
+    )
+      return;
     try {
       let { data } = rawEvent, { key, event, refId } = typeof data == "string" && isJSON(data) ? parse(data, CHANNEL_OPTIONS) : data;
       if (key === KEY) {
diff --git a/node_modules/storybook/dist/manager/globals-runtime.js b/node_modules/storybook/dist/manager/globals-runtime.js
index a154aaa..47ab6ca 100644
--- a/node_modules/storybook/dist/manager/globals-runtime.js
+++ b/node_modules/storybook/dist/manager/globals-runtime.js
@@ -30800,7 +30800,7 @@ var { document: document2, location: location2 } = globalThis, KEY = "storybook-
     );
     return frames.length ? (this.buffer.length && this.flush(), frames.forEach((f6) => {
       try {
-        f6.postMessage(data, "*");
+        f6.postMessage(data, globalThis.location ? globalThis.location.origin : "*");
       } catch {
         logger.error("sending over postmessage fail");
       }
@@ -30840,6 +30840,12 @@ var { document: document2, location: location2 } = globalThis, KEY = "storybook-
     ).flatMap((e2) => e2.contentWindow ? [e2.contentWindow] : []) : globalThis.parent ? [globalThis.parent] : [];
   }
   handleEvent(rawEvent) {
+    if (
+      rawEvent.origin &&
+      globalThis.location &&
+      rawEvent.origin !== globalThis.location.origin
+    )
+      return;
     try {
       let { data } = rawEvent, { key, event, refId } = typeof data == "string" && isJSON(data) ? parse(data, CHANNEL_OPTIONS) : data;
       if (key === KEY) {
diff --git a/node_modules/storybook/dist/preview/runtime.js b/node_modules/storybook/dist/preview/runtime.js
index 3cad961..cfcc6e6 100644
--- a/node_modules/storybook/dist/preview/runtime.js
+++ b/node_modules/storybook/dist/preview/runtime.js
@@ -12746,7 +12746,7 @@ var { document: document2, location } = globalThis, KEY = "storybook-channel", C
     );
     return frames.length ? (this.buffer.length && this.flush(), frames.forEach((f6) => {
       try {
-        f6.postMessage(data, "*");
+        f6.postMessage(data, globalThis.location ? globalThis.location.origin : "*");
       } catch {
         logger.error("sending over postmessage fail");
       }
@@ -12786,6 +12786,12 @@ var { document: document2, location } = globalThis, KEY = "storybook-channel", C
     ).flatMap((e2) => e2.contentWindow ? [e2.contentWindow] : []) : globalThis.parent ? [globalThis.parent] : [];
   }
   handleEvent(rawEvent) {
+    if (
+      rawEvent.origin &&
+      globalThis.location &&
+      rawEvent.origin !== globalThis.location.origin
+    )
+      return;
     try {
       let { data } = rawEvent, { key, event, refId } = typeof data == "string" && isJSON(data) ? parse(data, CHANNEL_OPTIONS) : data;
       if (key === KEY) {
